crypto/crypto/src/hash.rs

```diff
@@ -256,9 +256,10 @@ impl HashValue {
         })
     }
 
-    /// Returns first SHORT_STRING_LENGTH bytes as String in hex
-    pub fn short_str(&self) -> String {
-        hex::encode(&self.hash[..SHORT_STRING_LENGTH])
+    /// Returns first 4 bytes as hex-formatted string
+    pub fn short_str(&self) -> ShortHexStr {
+        // This will never fail since HashValue::LENGTH >= 4
+        ShortHexStr::from_bytes_unchecked(&self.hash)
```

Comment:
what's the point for these short hex strings? It's only for debugging?

Reply:
Comment probably needs to be updated; we use the short version for most (all?) logs and metrics that contain a peer id or account address. The full 32 chars "b50dd697e24abe597e718ae7f68246dc" is kind of unwieldy compared to just "b50dd697". Makes it a lot easier for humans to read imo.